extensions/amp-story/1.0/amp-story-interactive-results.js

```diff
@@ -0,0 +1,155 @@
+        <div class="i-amphtml-story-interactive-results-dots"></div>
+        <div class="i-amphtml-story-interactive-results-image-border">
+          <div class="i-amphtml-story-interactive-results-image"></div>
+        </div>
+        <div class="i-amphtml-story-interactive-results-dots"></div>
+      </div>
+      <div class="i-amphtml-story-interactive-results-prompt"></div>
+      <div class="i-amphtml-story-interactive-results-title"></div>
+      <div class="i-amphtml-story-interactive-results-description"></div>
+    </div>
+  `;
+};
+
+/**
+ * Processes the state and returns the condensed results.
+ * @param {!Map<string, {option: ?./amp-story-interactive.OptionConfigType, interactiveId: string}>} interactiveState
+ * @param {?Array<!./amp-story-interactive.OptionConfigType>} options needed to ensure the first options take precedence on ties
+ * @return {InteractiveResultsDef} the results
+ */
+const processResults = (interactiveState, options) => {
+  const categories = {};
+  // Add options in order to prefer earlier categories before later ones.
+  options.forEach((option) => {
+    categories[option.resultscategory] = 0;
+  });
+  Object.values(interactiveState).forEach((e) => {
+    if (e.option != null) {
+      categories[e.option.resultscategory] += 1;
+    }
+  });
+  return {
+    category: Object.keys(categories).reduce((a, b) =>
+      categories[a] >= categories[b] ? a : b
+    ),
+  };
+};
+
+export class AmpStoryInteractiveResults extends AmpStoryInteractive {
+  /**
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element, InteractiveType.RESULTS, [2, 4]);
+  }
+
+  /** @override */
+  buildCallback() {
+    super.buildCallback(CSS);
+  }
+
+  /** @override */
+  buildComponent() {
+    this.rootEl_ = buildResultsTemplate(this.element);
+    return this.rootEl_;
+  }
+
+  /** @override */
+  layoutCallback() {
+    if (this.element.hasAttribute('prompt-text')) {
+      this.rootEl_.querySelector(
+        '.i-amphtml-story-interactive-results-prompt'
+      ).textContent = this.element.getAttribute('prompt-text');
+    }
+    this.storeService_.subscribe(
+      StateProperty.INTERACTIVE_REACT_STATE,
+      (data) => {
+        const results = processResults(data, this.options_);
+        this.options_.forEach((e) => {
+          if (e.resultscategory === results.category) {
+            this.mutateElement(() => {
+              this.updateCategory_(e);
+              this.updateToPostSelectionState_(null);
+            });
+          }
+        });
+      },
+      true
+    );
+  }
+
+  /**
+   * Updates the element with the correct category
+   * @param {./amp-story-interactive.OptionConfigType} categorySelected
+   * @private
+   */
+  updateCategory_(categorySelected) {
+    setStyle(
+      this.rootEl_.querySelector('.i-amphtml-story-interactive-results-image'),
+      'background',
+      'url(' + categorySelected.image + ')'
```

Comment:
Do we need additional parameters like fit: cover/contain and no-repeat?

Reply:
I think it's a good idea to use cover for this case since we want the whole circle filled up, I'll add it 👍

Reply:
Actually the CSS already contains `background-size: cover !important; background-position: center !important;`. LMK if you think any other properties are needed